app/routes/admin/posts.py

```diff
@@ -153,6 +157,26 @@ def edit(post_id):
                            upload_form=upload_form)
 
 
+@posts.route('/posts/preview/<slug>')
+def preview(slug):
+    if BlogPost.objects(slug=slug).count() != 1:
+        abort(404)
+    post = BlogPost.objects().get(slug=slug)
+    if not post.date_published:
+        post.date_published = datetime.now()
```

Comment:
Gotcha.  Why don't you make a note of that, because it's pretty unclear.  Maybe just an inline comment indicating that the value isn't permanent.